pallets/proxy/src/lib.rs

```diff
@@ -0,0 +1,213 @@
+	use xcm::v2::{Junction::{Parachain, AccountId32}, SendXcm, Xcm, Junctions::{X1}};
+	use xcm::v2::prelude::*;
+	use frame_support::inherent::Vec;
+	// use bytes:str::ToBytes;
+
+	#[derive(Clone, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+	#[allow(non_camel_case_types)]
+	pub enum ProxyCall<T: Config> {
+		#[codec(index = 0u8)]
+		register { registration: JobRegistration<T::AccountId, T::RegistrationExtra> },
+
+		#[codec(index = 1u8)]
+		deregister { script: Script },
+
+		#[codec(index = 2u8)]
+		update_allowed_sources { script: Script, updates: Vec<AllowedSourcesUpdate<T::AccountId>> },
+
+		#[codec(index = 3u8)]
+		fulfill { fulfillment: Fulfillment, requester: <T::Lookup as StaticLookup>::Source },
+
+		#[codec(index = 4u8)]
+		submit_attestation { attestation_chain: AttestationChain },
+
+		#[codec(index = 5u8)]
+		update_certificate_revocation_list { update: CertificateRevocationListUpdate }
+	}
+
+	pub fn acurast_call<T: Config>(proxy: ProxyCall<T>, accountId: [u8; 32]) -> Result<(), SendError> {
+		let mut xcm_message = Vec::new();
+
+		// create an encoded version of the call composed of the first byte being the pallet id
+		// on the destination chain, second byte the position of the calling function on the enum,
+		// and then the arguments SCALE encoded in order
+		let mut encoded_call = Vec::<u8>::new();
+		encoded_call.push(T::AcurastPalletId::get() as u8);
+		encoded_call.append(&mut proxy.encode());
+		log::info!("encoded call is: {:?}", encoded_call);
+
+		// xcm_message.push(ClearOrigin);
+		// put our transact message in the vector of instructions
+		xcm_message.push(DescendOrigin(X1(AccountId32 {
+			network: NetworkId::Any,
+			id: accountId
+		})));
+
+		xcm_message.push(Transact {
+			origin_type: OriginKind::Xcm,
+			require_weight_at_most: 1_000_000_000 as u64,
+			call: encoded_call.into(),
+		});
+
+
+		// use router to send the xcm message
+		return match T::XcmSender::send_xcm(
+			(1, X1(Parachain(T::AcurastParachainId::get()))),
+			Xcm(xcm_message),
+		) {
+			Ok(_) => {
+				Ok(())
+			},
+			Err(e) => {
+				Err(e)
+			},
+		};
+	}
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// Extra structure to include in the registration of a job.
+		type RegistrationExtra: Parameter + Member + MaxEncodedLen + Eq;
+		type XcmSender: SendXcm;
+		type AcurastPalletId: Get<u8>;
+		type AcurastParachainId: Get<u32>;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event documentation should end with an array that provides descriptive names for event
+		/// parameters. [something, who]
+		SomethingStored(u32, T::AccountId),
+	}
```

Comment:
@JuaniRios Either emit proper events or we do not need to have events defined

Reply:
done